File: java/page.java
import com.splunk.*; 
import com.splunk.JobArgs.ExecutionMode;

import java.util.*;
import java.io.*;

public class page {

    public static void main(String[] args) {

		try {
			if(args.length != 1) {
				System.out.println("syntax: setmode \"search query\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        JobArgs ja = new JobArgs();
	        ja.setExecutionMode(ExecutionMode.BLOCKING);
			Job job = //LAB: create a job using the query in args[0] and ja

			int total = //LAB: get the total number of results from the job
			int offset = 0;
			int count = 5;
			while (offset < total) {
				System.out.println("====== PAGE " + (offset / count + 1) + " ============" );
				Args params = new Args();
				//LAB: add name/value pairs to params for count = count and offset = offset
				ResultsReaderXml set = //LAB: get the results from the job, passing params, and use them to create a new ResultsReaderXML
				Event event;       
				while ((event = set.getNextEvent()) != null) {
					System.out.println("------ RESULT " + //LAB: get the event's _serial field value + " -------------------");
					System.out.flush();
					for (String key: event.keySet())     
						System.out.print(key + "=" + event.get(key) + "| "); 
					System.out.println("--");
				}
				offset = offset + count; 
				System.in.read();
			}
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: python/page.py

import splunklib.client as client
import splunklib.results as results
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 2:
    print "Syntax: page.py \"search query\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)

    params = {"exec_mode": "blocking"}  #forces the call to create() to block until complete  
    job = #LAB: create the job using the query in sys.argv[1] and params
    
    total      = #LAB: get the total number of results in the job
    offset   = 0;                       
    pageSize = 10;                       
    
    print "Total results = %s" % str(total)

    while (offset < total):
        print "\n===== Page %s ===================================" % str(offset / pageSize + 1)
        pageParams = {#LAB: add name/value pairs for count=pageSize and offset=offset
        page = #LAB: get the job results, passing pageParams
        for result in results.ResultsReader(page):
            print "-----RESULT %s --------------------------------------" % #LAB: get the _serial field value from result 
            for key in result.keys():
                print "%s=%s|" % (key, result[key]),
            print 
        offset += pageSize
        raw_input('Enter to continue:')

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
